nixos/tests/statsd.nix

```diff
@@ -33,8 +33,19 @@ with lib;
   };
 
   testScript = ''
-    $statsd1->start();
-    $statsd1->waitForUnit("statsd.service");
-    $statsd1->waitUntilSucceeds("nc -z 127.0.0.1 8126");
+    $machine->start();
+    $machine->waitForUnit("statsd.service");
+    $machine->waitForOpenPort(8126);
+
+    # check state of the `statsd` server
+    $machine->succeed('[ "health: up" = "$(echo health | nc 127.0.0.1 8126 -w 120)" ];');
```

Comment:
You need to add `-N` for all TCP connections (but _not_ for UDP) so the connection is closed after EOF on input. Otherwise this will wait 120s even after a successful connection, and the test takes forever.